### PR TITLE
Fix/set the sync icon state while scan is pending

### LINKED ISSUE #254: [UI] Set the sync icon state while scan is pending
### Reason

The user should be able to see that a scan is both on going, and then that it has completed

### Description

When a scan is started, set the gmail and retailer states as SYNC if there are accounts linked and verified. On completion, the state should be reverted to P100.

### Success Criteria

- [ ] if there is a valid gmail account linked, show sync while scanning is ongoing
- [ ] if there is a valid retailer account linked, show sync while scanning is ongoing
- [ ] after sync is complete, if there is a valid gmail account, show state P100
- [ ] after sync is complete, if there is a valid retailer account, show state P100

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by MiroBenicio**
# Issues Addressed

Closes #254

### CODE REVIEW

**Comment by MiroBenicio** [comment]
@mike-audi 

PR updated with requested changes

File: src/components/sheet/sheet-add-gmail.vue [hunk 1]
```diff
@@ -36,10 +38,14 @@ const submit = async () => {
...
     await capture.login(form.value);
+    await store.gmail.set(BulletState.SYNC)
     form.value = { username: "", password: "", type: GMAIL };
-    capture.scan().catch((error) => console.error(error.toString()));
+    capture.scan().catch((error) => console.error(error.toString())).finally(async ()=>
+      await store.gmail.set(BulletState.P100)
+    );
     emit("back");
   } catch (err: any) {
+    await store.gmail.set(BulletState.NULL)
```
Comment (mike-audi) [inline src/components/sheet/sheet-add-gmail.vue:48]: If there is an error, it should be set to `BulletState.Error` not Null

Comment (mike-audi) [inline src/components/sheet/sheet-add-gmail.vue:48]: If there is an error, it should be set to BulletState.Error not Null

File: src/components/sheet/sheet-add-retailer.vue [hunk 2]
```diff
@@ -66,6 +72,7 @@ const submit = async () => {
     };
     emit("back");
   } catch (err: any) {
+    await store.retailer.set(BulletState.NULL)
```
Comment (mike-audi) [inline src/components/sheet/sheet-add-retailer.vue:75]: If there is an error, it should be set to BulletState.Error not Null

**Review by mike-audi — CHANGES_REQUESTED** [review]

File: src/components/sheet/sheet-add-gmail.vue (hunk 1 above)
Comment (MiroBenicio) [reply src/components/sheet/sheet-add-gmail.vue]: Changed

File: src/components/sheet/sheet-add-retailer.vue (hunk 2 above)
Comment (MiroBenicio) [reply src/components/sheet/sheet-add-retailer.vue]: Changed

**Review by mike-audi — APPROVED** [review]

**Pull request merged**